internal/installercache/installercache_test.go

```diff
@@ -53,60 +59,81 @@ var _ = Describe("release event", func() {
+		ctx             context.Context
+		diskStatsHelper metrics.DiskStatsHelper
 	)
 
 	BeforeEach(func() {
-		DeleteGracePeriod = 1 * time.Millisecond
 
 		ctrl = gomock.NewController(GinkgoT())
+		diskStatsHelper = metrics.NewOSDiskStatsHelper(logrus.New())
 		mockRelease = oc.NewMockRelease(ctrl)
 		eventsHandler = eventsapi.NewMockHandler(ctrl)
-
 		var err error
 		cacheDir, err = os.MkdirTemp("/tmp", "cacheDir")
 		Expect(err).NotTo(HaveOccurred())
 		Expect(os.Mkdir(filepath.Join(cacheDir, "quay.io"), 0755)).To(Succeed())
 		Expect(os.Mkdir(filepath.Join(filepath.Join(cacheDir, "quay.io"), "release-dev"), 0755)).To(Succeed())
-		manager = New(cacheDir, 12, eventsHandler, logrus.New())
+		installerCacheConfig := Config{
+			CacheDir:                  cacheDir,
+			MaxCapacity:               Size(12),
+			MaxReleaseSize:            Size(5),
+			EvictionThresholdPercent:  100,
+			ReleaseFetchRetryInterval: 1 * time.Microsecond,
+		}
+		manager, err = New(installerCacheConfig, eventsHandler, diskStatsHelper, logrus.New())
+		Expect(err).NotTo(HaveOccurred())
 		ctx = context.TODO()
 	})
 
 	AfterEach(func() {
-		os.RemoveAll(cacheDir)
+		//os.RemoveAll(cacheDir)
 	})
 
-	expectEventSent := func() {
+	expectEventsSent := func() {
 		eventsHandler.EXPECT().V2AddMetricsEvent(
 			ctx, gomock.Any(), nil, nil, "", models.EventSeverityInfo, metricEventInstallerCacheRelease, gomock.Any(),
 			gomock.Any(),
-		).Times(1)
+		).AnyTimes()
 	}
 
-	testGet := func(releaseID, version string, clusterID strfmt.UUID, expectCached bool) (string, string) {
-		workdir := filepath.Join(cacheDir, "quay.io", "release-dev")
+	mockReleaseCalls := func(releaseID string, version string) {
+		workdir := filepath.Join(manager.config.CacheDir, "quay.io", "release-dev")
 		fname := filepath.Join(workdir, releaseID)
 
 		mockRelease.EXPECT().GetReleaseBinaryPath(
 			gomock.Any(), gomock.Any(), version).
-			Return(workdir, releaseID, fname, nil)
+			Return(workdir, releaseID, fname, nil).AnyTimes()
+
 		mockRelease.EXPECT().Extract(gomock.Any(), releaseID,
-			gomock.Any(), cacheDir, gomock.Any(), version).
+			gomock.Any(), manager.config.CacheDir, gomock.Any(), version).
 			DoAndReturn(func(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, cacheDir string, pullSecret string, version string) (string, error) {
-				err := os.WriteFile(fname, []byte("abcde"), 0600)
+				dir, _ := filepath.Split(fname)
+				err := os.MkdirAll(dir, 0700)
+				Expect(err).ToNot(HaveOccurred())
+				time.Sleep(10 * time.Millisecond) // Add a small amount of latency to simulate extraction
+				err = os.WriteFile(fname, []byte("abcde"), 0600)
 				return "", err
-			})
+			}).AnyTimes()
+	}
+
+	testGet := func(releaseID, version string, clusterID strfmt.UUID, expectCached bool) (string, string) {
+		workdir := filepath.Join(manager.config.CacheDir, "quay.io", "release-dev")
+		fname := filepath.Join(workdir, releaseID)
+		mockReleaseCalls(releaseID, version)
```

Comment:
Should this depend on `expectCached` as well?

Reply:
I am not sure what you mean exactly? 

But the intent here is to be able to pass expectations about the cached status within the test so that assertions may be made.

It seems to be doing the job?

Reply:
I guess I'm asking if we want to ensure we're not calling the release code if we're expecting the release to be cached.

Reply:
I mean, TBH, calling `release.Cleanup` is not a huge issue, we need to remember that this is just cleaning up the hardlinks to simulate a normal flow of the system.

Even if something is cached, a hardlink is issued every time someone requests that release via `Get`

Clearing the hardlink may assist the eviction of that release in some cases (as there will be no hardlinks) but I don't think any of the cases in `testGet` are doing anything like this.

Reply:
I don't understand this comment.. This line has nothing to do with `Cleanup`

`mockReleaseCalls` is saying "I'm expecting this release to be extracted", but we shouldn't expect that if the release should be cached, right?

Reply:
Oh sorry, I misunderstood your original statement, you mean to check we are not calling the 'release extraction' code. Got it! I'll look at this.